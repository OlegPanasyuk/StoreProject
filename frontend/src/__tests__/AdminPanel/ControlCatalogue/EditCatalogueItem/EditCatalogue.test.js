import React from 'react';
import
EditCatalogueItemConnected,
{ EditCatalogueItem }
    from '../../../../AdminPanel/ControlCatalogue/EditCatalogueItem';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    getCatalogue: () => { },
    cancel: () => { },
    addErrorToState: () => { },
    editItem: {
        show: false,
        id_catalogue: 1,
        name: '',
        description: '',
        parent_id: ''
    }
}) {
    let component = shallow(<EditCatalogueItem {...props} />);
    return { component, props };
}

describe('EditCatalogueItem --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(EditCatalogueItem.propTypes.editItem).toBe(PropTypes.object);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});

describe('EditCatalogueItem ', () => {
    let initialState = {
        adminPanel_catalogue: {
            editItem: {
                show: false,
                id_catalogue: 1,
                name: 'nnn',
                description: 'ddd',
                parent_id: -1
            }
        }
    };

    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><EditCatalogueItemConnected /></Provider>);
    });
    it('+++ render the connected(SMART) component',() => {
        expect(wrapper.find(EditCatalogueItemConnected).length).toEqual(1);
    });
});
