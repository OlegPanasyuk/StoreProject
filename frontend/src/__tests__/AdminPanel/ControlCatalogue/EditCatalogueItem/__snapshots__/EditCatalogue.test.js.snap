// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCatalogueItem --- Snapshot Component must be rendered 1`] = `
<Fragment>
  <ConnectFunction
    getCatalogue={[Function]}
    onHide={[Function]}
    show={false}
  />
  <Form
    as="form"
    inline={false}
  >
    <h4>
      Edit Item
    </h4>
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Name:
      </FormLabel>
      <FormControl
        as="input"
        defaultValue=""
        type="text"
      />
    </FormGroup>
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        description:
      </FormLabel>
      <FormControl
        as="input"
        defaultValue=""
        type="text"
      />
    </FormGroup>
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Parent:
      </FormLabel>
      <FormControl
        as="input"
        defaultValue=""
        type="number"
      />
    </FormGroup>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Save
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Cancel
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Delete
    </Button>
  </Form>
</Fragment>
`;
